godot-core/src/builtin/plane.rs

```diff
@@ -191,6 +310,301 @@ mod test {
+        assert_eq!(plane.intersects_segment(vec_b, vec_c), None);
+        assert_eq!(plane.intersects_segment(vec_b, vec_g), None);
+    }
+
+    /// Tests `is_equal_approx()`.
+    #[test]
+    fn test_equal() {
+        // Initial normalized planes.
+        let plane_a: Plane = Plane::new(Vector3::new(0.0, 0.0, 1.0).normalized(), 0.0);
+        let plane_b: Plane = Plane::new(Vector3::new(0.01, 0.0, 1.0).normalized(), 0.0);
+        let plane_c: Plane = Plane::new(Vector3::new(0.0001, 0.0, 1.0).normalized(), 0.0);
+        let plane_d: Plane = Plane::new(Vector3::new(0.000001, 0.0, 1.0).normalized(), 0.0);
+        let plane_e: Plane = Plane::new(Vector3::new(0.000001, 0.0, 1.0).normalized(), 0.01);
+        let plane_f: Plane = Plane::new(Vector3::new(0.000001, 0.0, 1.0).normalized(), 0.000001);
+
+        // Same planes should be equals.
+        assert_eq_approx!(&plane_a, &plane_a, Plane::is_equal_approx);
+        assert_eq_approx!(&plane_b, &plane_b, Plane::is_equal_approx);
+        // Although similar, planes below shouldn't be approximately equals.
+        assert_ne_approx!(&plane_a, &plane_b, Plane::is_equal_approx);
+        assert_ne_approx!(&plane_a, &plane_c, Plane::is_equal_approx);
+        // Planes below should be approximately equal because it's lower than the set tolerance constant.
+        assert_eq_approx!(&plane_a, &plane_d, Plane::is_equal_approx);
+        // Although approximately equal in the `normal` part, it is not approximately equal in the `d`
+        // part.
+        assert_ne_approx!(&plane_a, &plane_e, Plane::is_equal_approx);
+        // Both attributes are approximately equal.
+        assert_eq_approx!(&plane_a, &plane_f, Plane::is_equal_approx);
+        // Although considered approximately equal with `plane_a`, `plane_b` is not considered approximately
+        // equal with `plane_e` because the baseline comparison is tighter.
+        assert_ne_approx!(&plane_b, &plane_e, Plane::is_equal_approx);
+    }
+
+    /// Tests `normalize()`.
+    #[test]
+    fn test_normalization() {
+        // Initial normalized planes (Denormalization not done here because the constructor panics
+        // when creating a plane with unnormalized planes).
+        let mut plane_a: Plane = Plane::new(Vector3::new(0.7, 2.0, 6.0).normalized(), 0.0);
+        let mut plane_b: Plane = Plane::new(Vector3::new(1.5, 7.2, 9.1).normalized(), 2.0);
+        let mut plane_c: Plane = Plane::new(Vector3::new(1.4, 9.1, 1.2).normalized(), 5.3);
+        let mut plane_d: Plane = Plane::new(Vector3::new(4.2, 2.9, 1.5).normalized(), 2.4);
+        let plane_e: Plane = Plane::new(Vector3::new(5.1, 3.0, 2.1).normalized(), 0.2);
+
+        // Denormalize planes.
+        plane_a.normal = Vector3::new(0.7, 2.0, 6.0);
+        plane_b.normal = Vector3::new(1.5, 7.2, 9.1);
+        plane_c.normal = Vector3::new(1.4, 9.1, 1.2);
+        plane_d.normal = Vector3::new(4.2, 2.9, 1.5);
```

Comment:
The normals above are overwritten without ever being used.

Why not just create the planes with the struct syntax `Plane { normal: ..., d: ... }`? 
Needs half the code and avoids `mut` as well.